File: API/API/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace API.Models
{
    public class Product
    {
        [Key]
        public int Id { get; set; }
        public string Image { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Subtitle { get; set; }
        public double Price { get; set; }
        public string CategoryName { get; set; }
        public Category Category { get; set; }
    }
}

File: online_shop/lib/models/Product.dart
class Product{
  final String image, title, description, subtitle;
  final int id;
  final double price;
  Product({
    this.subtitle, 
    this.id,
    this.image,
    this.title,
    this.price,
    this.description,
  });

  factory Product.fromJson(Map<String, dynamic> data){
    return Product(
      id: data["id"],
      image: data["image"],
      title: data["title"],
      price: data["price"],
      description: data["description"],
      subtitle: data["subtitle"]
    );
  }
}